rust/core-lib/src/view.rs

```diff
@@ -486,6 +480,11 @@ impl View {
         let height = self.line_of_offset(text, text.len()) + 1;
         let plan = RenderPlan::create(height, self.first_line, self.height);
         self.send_update_for_plan(text, tab_ctx, style_spans, &plan);
+        if let Some(new_scroll_pos) = self.scroll_to {
```

Comment:
It's tempting to write `self.scroll_to.take() and save a line, but this is probably easier to understand.